weather-wise/src/main/java/app/weatherwise/Controller.java

```diff
@@ -16,40 +14,84 @@ public class Controller {
     private final String query = "q="; // parameter to specify city name
     private final String and = "&"; // syntax for adding parameters
     private final String noData = "NOT FOUND!"; // null replacement
-    private final double noDouble = 404;
+    private final double noDouble = 404; // null replacement
 
     private final String apiKey = "3d3d6e9a8b385dea9f19a7323938d980"; // API key to authenticate call
+    private APIService service; // instantiate APIService object
 
     @FXML
-    private TextField latitudeInput, longitudeInput, cityInput;
+    private RadioButton selectCity, selectGeolocation;
     @FXML
-    private Label output;
+    private Label latitudeText, longitudeText, cityText, output;
     @FXML
-    private ChoiceBox<String> textInputBox;
+    private TextField latitudeInput, longitudeInput, cityInput;
     @FXML
-    private final String[] textInputTypes = {"Geolocation", "City"};
+    private Button button;
 
     @FXML
-    public void initialize() {
-        textInputBox.getItems().addAll(textInputTypes);
-        textInputBox.setValue(textInputTypes[0]);
+    private void initialize() {
+        service = APIService.getInstance(); // create or get the APIService instance
+        selectCity.setSelected(true);
+        latitudeText.setVisible(false);
+        longitudeText.setVisible(false);
+        latitudeInput.setVisible(false);
+        longitudeInput.setVisible(false);
+        button.setDefaultButton(true);
+    }
+
+    public void switchInputMode(){
+        if(selectCity.isSelected()){
+            output.setText("");
+            cityText.setVisible(true);
+            cityInput.setVisible(true);
+            latitudeText.setVisible(false);
+            longitudeText.setVisible(false);
+            latitudeInput.setVisible(false);
+            longitudeInput.setVisible(false);
+        } else {
+            output.setText("");
+            cityText.setVisible(false);
+            cityInput.setVisible(false);
+            latitudeText.setVisible(true);
+            longitudeText.setVisible(true);
+            latitudeInput.setVisible(true);
+            longitudeInput.setVisible(true);
+        }
```

Comment:
Implemented the effect of switching "scenes", between selecting City and Geolocation where two different input prompts are shown per selection.